Add lancedb-haystack to integrations.

alanmeeson opened the pull request:
This PR adds the LanceDB Haystack Document Store from [https://github.com/alanmeeson/lancedb-haystack](https://github.com/alanmeeson/lancedb-haystack). 

Notes from Discord discussion:
There are some interesting quirks due to the nature of LanceDB; you need to specify a schema for your metadata so that it can be turned into a pyarrow struct (and thus be searchable).  Also the precision of the embedding vectors seems to be limited to an error of around 10^-7 compared to it's python form - feels like some sort of difference between the float representations. 

Please take a look when you have time, and let me know if I should change anything.

Thanks!

bilgeyucel commented on integrations/lancedb.md:
````diff
@@ -0,0 +1,66 @@
+
+The current simplest way to get LanceDB-Haystack is to install from GitHub via pip:
+
+```pip install lancedb-haystack```
````
```suggestion
```shell
pip install lancedb-haystack

```

bilgeyucel replied:
It's quite hard to show it with github's markdown syntax. This is what I mean
![image](https://github.com/deepset-ai/haystack-integrations/assets/25897116/47280383-f192-4349-97af-9b0f7a9570d4)

alanmeeson replied:
Updated :heavy_check_mark:

bilgeyucel reviewed (commented):
Thanks for the PR @alanmeeson. I left one minor comment. Also feel free to add a logo that will be displayed on a dark background 😊

alanmeeson commented:
I've updated the pip install block to be tagged as shell,  got in touch with the LanceDB team and sourced the official 'light on dark' logo, and added a link to the new API docs.

bilgeyucel reviewed (approved):
🚀
